src/components/ChatRoom/MessengerContainer.js

```diff
@@ -0,0 +1,201 @@
+        text: text,
+        date: new Date().getTime(),
+        isHeart: false,
+      });
+      setText('');
+      setChatEdit(!chatEdit);
+    }
+  };
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    handleSubmitButtonClick();
+  };
+  const onChange = (e) => {
+    const nextForm = e.target.value;
+    setText(nextForm);
+  };
+
+  const onHeartButtonClick = () => {
+    console.log('heart 누름');
+    // heart submit해야함.
+    setText('');
+    onSubmit({
+      id: isMe,
+      text: text,
+      date: new Date().getTime(),
+      isHeart: true,
+    });
+    setChatEdit(!chatEdit);
+  };
+  // 나인지, 상대인지 토글
+  function onClickToggle() {
+    setIsMe(!isMe);
+  }
+
+  function renderForm() {
+    return (
+      <MessengerTextFormContainer>
+        <Form onSubmit={handleSubmit}>
+          <Button paddingLeft="20px">
+            <img
+              className="icon"
+              src="img/emoji.png"
+              width="24px"
+              alt="emoji"
+            />
+          </Button>
+          <TextInput
+            type="text"
+            placeholder="메시지 입력..."
+            onChange={onChange}
+            value={text}
+          />
+          <Button>
+            <img
+              className="icon"
+              src="img/gallery.png"
+              width="24px"
+              alt="emoji"
+            />
+          </Button>
+          <Button onClick={onHeartButtonClick}>
+            <img
+              className="icon"
+              src="img/heart.png"
+              width="24px"
+              alt="emoji"
+            />
+          </Button>
+        </Form>
+      </MessengerTextFormContainer>
+    );
+  }
+
+  function renderHeart() {
+    return (
+      <HeartContainer>
+        <HeartImg
+          alt="profile-img"
+          src="img/redHeart.png"
+          width="44px"
+          height="44px"
+        />
+      </HeartContainer>
+    );
+  }
+
+  function renderMessages() {
+    return messengerData.chatData.map((element, index) => {
+      return (
+        <Chat isMe={element.isMe} key={index}>
+          {renderProfile(element.isMe)}
+          {element.isHeart ? (
+            renderHeart()
+          ) : (
+            <ChatBubble>
+              <TextBox>{element.text}</TextBox>
+            </ChatBubble>
+          )}
+          <TimeBox>{dateToString(element.date)}</TimeBox>
+        </Chat>
+      );
+    });
+  }
+
+  function renderProfile(prop, isTop) {
+    return (
+      <>
+        {!prop ? (
+          <>
+            <ProfileImage
+              alt="profile-img"
+              src="img/1.png"
+              width="22px"
+              height="22px"
+            />
+            <>{isTop ? <h4>__kiuk</h4> : <></>}</>
+          </>
+        ) : (
+          <>
+            <ProfileImage
+              alt="profile-img"
+              src="img/noImg.png"
+              width="22px"
+              height="22px"
+            />
+            <>{isTop ? <h4>n0wkim</h4> : <></>}</>
```

Comment:
```suggestion
            <>{isTop && <h4>__kiuk</h4>}</>
          </>
        ) : (
          <>
            <ProfileImage
              alt="profile-img"
              src="img/noImg.png"
              width="22px"
              height="22px"
            />
            <>{isTop && <h4>n0wkim</h4>}</>
```
[단축 평가 논리 계산법](https://velog.io/@nudge411/%EB%8B%A8%EC%B6%95-%ED%8F%89%EA%B0%80-short-circuit-evaluation-%EB%85%BC%EB%A6%AC-%EA%B3%84%EC%82%B0%EB%B2%95)
삼항연산자 말고 이렇게 써 보는 건 어떨까요?!